### PR TITLE
Added Largest Series Product Generator

### PR DESCRIPTION
**Pull request by felix91gr**
- Changed name of Property to `GetLargestProduct`, to fit with example and stub implementation
- Fixed issues in example implementation, that didn't handle the newest tests' requirements. In particular, erroneous input crashed the example, while the testing suite **expected it to return -1**. Now the example behaves properly.
- Updated `eol` to canonical (unintended, but the generator created it).

### CODE REVIEW

File: exercises/largest-series-product/Example.cs [hunk 1]
```diff
@@ -22,6 +36,7 @@ private static long ParseDigit(char c)
     {
         if (!char.IsDigit(c))
         {
+            Console.WriteLine("Invalid digit: " + c);
```
Comment (ErikSchierboom) [inline exercises/largest-series-product/Example.cs]: Could you remove this line?

File: exercises/largest-series-product/LargestSeriesProductTest.cs [hunk 2]
```diff
@@ -1,171 +1,96 @@
...
+    [Fact(Skip = "Remove to run test")]
+    public void Reports_1_for_nonempty_string_and_empty_product_0_span_()
+    {
+        Assert.Equal(1, LargestSeriesProduct.GetLargestProduct("123", 0));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Rejects_empty_string_and_nonzero_span()
+    {
+        Assert.Equal(-1, LargestSeriesProduct.GetLargestProduct("", 1));
```
Comment (ErikSchierboom) [inline exercises/largest-series-product/LargestSeriesProductTest.cs]: The `-1` is an error case, and should throw an exception. Same goes for the other `-1` cases.

File: generators/Exercises/LargestSeriesProduct.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+using Generators.Input;
+
+namespace Generators.Exercises
+{
+    public class LargestSeriesProduct : Exercise
+    {
+        protected override void UpdateCanonicalData(CanonicalData canonicalData)
+        {
+            foreach (var canonicalDataCase in canonicalData.Cases)
+                canonicalDataCase.Property = "GetLargestProduct";
```
Comment (ErikSchierboom) [inline generators/Exercises/LargestSeriesProduct.cs]: Maybe `CalculateLargestSeriesProduct` would be a better name.

File: exercises/largest-series-product/Example.cs (hunk 1 above)
Comment (felix91gr) [reply exercises/largest-series-product/Example.cs]: Yes!

File: exercises/largest-series-product/LargestSeriesProductTest.cs (hunk 2 above)
Comment (felix91gr) [reply exercises/largest-series-product/LargestSeriesProductTest.cs]: So, if I'm understanding this, a `-1` in the `canonical-data.json` represents an exception?

File: generators/Exercises/LargestSeriesProduct.cs (hunk 3 above)
Comment (felix91gr) [reply generators/Exercises/LargestSeriesProduct.cs]: Hmm. Maybe?

I interpret `GetLargestProduct` as *I wanna get the largest product in this series*, which fits the exercise description. Basically the same as saying `CalculateLargestSeriesProduct`, but more precise. I would leave it, but regardless, both convey the correct meaning.

File: exercises/largest-series-product/LargestSeriesProductTest.cs (hunk 2 above)
Comment (felix91gr) [reply exercises/largest-series-product/LargestSeriesProductTest.cs]: Ah, yes. Indeed it does:

negative numbers in canonical data

File: exercises/largest-series-product/Example.cs (hunk 1 above)
Comment (felix91gr) [reply exercises/largest-series-product/Example.cs]: I reverted all of my changes to `Example.cs` :)

File: exercises/largest-series-product/LargestSeriesProductTest.cs (hunk 2 above)
Comment (felix91gr) [reply exercises/largest-series-product/LargestSeriesProductTest.cs]: I added the `Assert.Throws`!

**Comment by felix91gr** [comment]
I think I've fixed all of the issues. Thanks for the feedback!

File: generators/Exercises/LargestSeriesProduct.cs (hunk 3 above)
Comment (ErikSchierboom) [reply generators/Exercises/LargestSeriesProduct.cs]: Well argued. I agree

**Comment by ErikSchierboom** [comment]
Thanks!

**Comment by felix91gr** [comment]
:boat: :D

**Pull request merged**